src/core/Akka.Streams/Implementation/SubscriberManagement.cs

```diff
@@ -313,9 +313,10 @@ protected void PushToDownstream(T value)
             if (_endOfStream is SubscriberManagement.NotReached)
             {
                 _pendingFromUpstream--;
+                var oldBufferLength = _buffer.Value.Length;
                 if (!_buffer.Value.Write(value))
                     throw new IllegalStateException("Output buffer overflow");
-                if (Dispatch(_subscriptions))
+                if (_buffer.Value.Length > oldBufferLength && Dispatch(_subscriptions))
```

Comment:
How does this work exactly? Wouldn't we just want to pull so long as there's demand?

Reply:
Good catch, that's not supposed to be there. I'm fixing it right now.